### PR TITLE
refactor: less state dependent

### PR DESCRIPTION
**Pull request by samedii**

### CODE REVIEW

File: kw6/reader.py [hunk 1]
```diff
@@ -46,21 +47,26 @@ def from_path(path: Path, header_path: Path = None):
         if version != "KW6FileClassVer1.0":
             raise ValueError(f"Unexpected file version {version}")
 
-        initial_position_header = Position.skip_(stream)
+        initial_position_header = PositionHeader.from_stream_(stream)
+
+        cached_byte_positions = (
+            hdr.positions(header_path)
```
Comment (FelixAbrahamsson) [inline kw6/reader.py]: I realize now that this was not part of this PR but isnt `header` a more descriptive name than `hdr`? That's what we call it in kmm

Comment (samedii) [reply kw6/reader.py]: Sure

Comment (samedii) [reply kw6/reader.py]: But we also have two other "headers" already so I'm not sure it's great idea actually

File: kw6/reader.py [hunk 2]
```diff
@@ -73,44 +79,29 @@ def __iter__(self):
...
-        if position.header.frame_index != expected_frame_index:
-            raise Exception(
-                f"Expected to find {expected_frame_index}, "
-                f"but found {position.header.frame_index} instead."
-            )
-        return assumptuous_length
+        from_position = self[from_frame_index]
+        max_byte_position = self.cached_byte_positions[from_frame_index]
+        return int(
+            (self.n_bytes - max_byte_position) / np.clip(from_position.header.n_frame_bytes, 100, 400000)
```
Comment (FelixAbrahamsson) [inline kw6/reader.py:102]: Why is `np.clip` needed here?

Comment (samedii) [reply kw6/reader.py:102]: It's more of an artifact that I forgot about though it's not entirely unreasonable. I'll remove it
